lib/wab/impl/configuration.rb

```diff
@@ -1,105 +1,228 @@
-          # TBD: Employ Oj or builtin JSON to load file.
-        when /\.ya?ml/i
+        when /\.json$/i
+          Oj.load_file(file, mode: :strict, symbol_keys: true)
+        when /\.ya?ml$/i
           begin
             require 'safe_yaml/load'
             SafeYAML.load_file(file) || {}
           rescue LoadError
-            puts "Could not load the requested resource.\n" \
-              "Please install the 'safe_yaml' gem via Bundler or directly, " \
-              "and try loading again.."
-            {}
+            # Re-raise with a more descriptive message. This should generally
+            # abort the configuration loading.
+            raise LoadError.new(%{Could not load the requested resource. Please install the 'safe_yaml' gem via
+Bundler or directly, and try loading again.
+})
           end
         end
       end
 
       # Returns a Hash containing data obtained by parsing a UNIX style conf
-      # file. Currently, only a three-tier nesting is supported, while higher
-      # tiers are silently ignored.
+      # file.
       #
       # For example, +handler.sample.count = 63+ and +handler.sample.path = /v1+
       # will be parsed into the following:
       #
-      #    {"handler"=>{"sample"=>{"count"=>63, "path"=>"/v1"}}}
-      #
-      # but +handler.sample.path.node = Article+ will be ignored.
+      #    { handler: { sample: { count: 63, path: "/v1" } } }
       def parse_conf_file(file)
-        # support nesting hashes upto three-levels deep only
-        config = Hash.new do |hsh, key|
-          hsh[key] = Hash.new do |h, k|
-            h[k] = Hash.new(&:default_proc)
-          end
-        end
+        config = {}
 
         File.open(File.expand_path(file)) do |f|
           f.each_line do |line|
             line.strip!
             next if line.empty? || line.start_with?('#')
             key, value = line.split('=').map(&:strip)
+            set_map(config, key, value)
+          end
+        end
+        config
+      end
 
-            keys = key.split(".")
-            primary_key = keys.shift
-            if keys.empty?
-              config[primary_key] = value
+      def set_map(node, path, value)
+        return node if path.empty?
+        path = path.to_s.split('.') unless path.is_a?(Array)
+        path[0..-2].each_index { |i|
+          key = path[i]
+          if node.is_a?(Hash)
+            key = key.to_sym
+            unless node.has_key?(key)
+              ai = Utils.key_to_int_ok(path[i + 1])
+              node[key] = ai.nil? ? {} : []
+            end
+            node = node[key]
+          elsif node.is_a?(Array)
+            key = Utils.key_to_int(key)
+            if key < node.length && -node.length < key
+              node = node[key]
             else
-              secondary_key = keys.shift
-              if keys.empty?
-                config[primary_key][secondary_key] = value
+              ai = Utils.key_to_int_ok(path[i + 1])
+              nn = ai.nil? ? {} : []
```

Comment:
please rename to a more descriptive local variables `ai` and `nn` just doesn't cut it..

Reply:
The variables are used within a block of 8 lines and the `ai` is used just one line apart. Any developer would be able to tell where `nn` is used. Not every local variables has to tell a story with a long name.

I'll solve the `ai` name by getting rid of it.